core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/geodistance/GeoDistanceParser.java

```diff
@@ -164,6 +162,13 @@ public AggregatorFactory parse(String aggregationName, XContentParser parser, Se
         return new GeoDistanceFactory(aggregationName, vsParser.config(), InternalGeoDistance.FACTORY, origin, unit, distanceType, ranges, keyed);
     }
 
+    private static String getSearchParseExceptionMessage(String currentFieldName, String aggregationName) {
+        return "Failed when parsing field with name '" +  currentFieldName +
+                "' found inside aggregation with name '" + aggregationName + "'." + " Few Possible Causes for the error:" +
+                "1)Misspelling the field name.  2)Giving a field name which is not recognized by Geo Distance Aggregator." +
+                "3)Assigning the field to a value with unallowed object type. ";
```

Comment:
Enumerating all possible clauses of a parsing failure doesn't look helpful to me. I think we just need to include the field name in the exception that we create on line 148?